linux/scripts/deb-packaging.sh

```diff
@@ -50,19 +56,151 @@ source_action() {
   mv builddebs/* "${OUTPUT_PATH:-..}"
 }
 
+output_log() {
+  echo "$1" >&2
+  builder_echo "$1"
+}
+
+output_ok() {
+  echo ":heavy_check_mark: $1" >&2
+  builder_echo green "OK: $1"
+}
+
+output_warning() {
+  echo ":warning: $1" >&2
+  builder_echo warning "WARNING: $1"
+}
+
+output_error() {
+  echo ":x: $1" >&2
+  builder_echo error "ERROR: $1"
+}
+
+check_api_not_changed() {
+  # Checks that the API did not change compared to what's documented in the .symbols file
+  tmpDir=$(mktemp -d)
+  dpkg -x "${BIN_PKG}" "${tmpDir}"
+  cd debian
+  dpkg-gensymbols -v"${PKG_VERSION}" -p"${PKG_NAME}" -e"${tmpDir}"/usr/lib/x86_64-linux-gnu/"${LIB_NAME}".so* -O"${PKG_NAME}.symbols" -c4
+  output_ok "${LIB_NAME} API didn't change"
+  cd "${REPO_ROOT}/linux"
+}
+
+is_symbols_file_changed() {
+  local CHANGED_REF CHANGED_BASE
+  CHANGED_REF=$(git rev-parse "${GIT_REF}":"linux/debian/${PKG_NAME}.symbols")
+  CHANGED_BASE=$(git rev-parse "${GIT_BASE}":"linux/debian/${PKG_NAME}.symbols")
+  if [[ "${CHANGED_REF}" == "${CHANGED_BASE}" ]]; then
+    return 1
+  fi
+  return 0
+}
+
+check_updated_version_number() {
+  # Checks that the package version number got updated in the .symbols file if it got changed
+  # shellcheck disable=SC2310
+  if is_symbols_file_changed; then
+    # .symbols file changed, now check if the package version got updated as well
+    # Note: We don't check that ALL changes in that file have an updated package version -
+    # we hope this gets flagged in code review.
+    if ! git log -p -1 -- "debian/${PKG_NAME}.symbols" | grep -q "${PKG_VERSION}"; then
```

Comment:
`${PKG_VERSION}` could just be `${VERSION}` I think? And then we can eliminate all of that from this script?